readthedocs/rtd_tests/tests/test_middleware.py

```diff
@@ -111,28 +113,29 @@ def test_request_header(self):
 
     @override_settings(PRODUCTION_DOMAIN='readthedocs.org')
     def test_proper_cname_uppercase(self):
-        cache.get = lambda x: x.split('.')[0]
+        get(Domain, project=self.pip, domain='pip.random.com')
         request = self.factory.get(self.url, HTTP_HOST='PIP.RANDOM.COM')
         self.middleware.process_request(request)
         self.assertEqual(request.urlconf, self.urlconf_subdomain)
         self.assertEqual(request.cname, True)
         self.assertEqual(request.slug, 'pip')
 
     def test_request_header_uppercase(self):
-        request = self.factory.get(self.url, HTTP_HOST='some.random.com', HTTP_X_RTD_SLUG='PIP')
+        request = self.factory.get(
+            self.url, HTTP_HOST='some.random.com', HTTP_X_RTD_SLUG='PIP'
+        )
         self.middleware.process_request(request)
         self.assertEqual(request.urlconf, self.urlconf_subdomain)
         self.assertEqual(request.cname, True)
         self.assertEqual(request.rtdheader, True)
         self.assertEqual(request.slug, 'pip')
 
-    @override_settings(USE_SUBDOMAIN=True)
-    # no need to do a real dns query so patch cname_to_slug
-    @patch('readthedocs.core.middleware.cname_to_slug', new=lambda x: 'doesnt')
-    def test_use_subdomain_on(self):
-        request = self.factory.get(self.url, HTTP_HOST='doesnt.really.matter')
-        ret_val = self.middleware.process_request(request)
-        self.assertIsNone(ret_val, None)
+    def test_use_subdomain(self):
+        domain = 'doesnt.exists.org'
+        get(Domain, project=self.pip, domain=domain)
+        request = self.factory.get(self.url, HTTP_HOST=domain)
+        res = self.middleware.process_request(request)
+        self.assertIsNone(res)
```

Comment:
I think it worth to check here that `request.slug` is `pip` and `request.domain_object` is `True`, otherwise we are not sure that we are picking the right project.